index.d.ts

```diff
@@ -0,0 +1,189 @@
+/**
+ * This file was automatically generated by json-schema-to-typescript.
+ * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
+ * and run json-schema-to-typescript to regenerate this file.
+ */
+
+export type ErrorCode = string;
+export type CloudFunctionRuntime =
+  | 'nodejs6'
+  | 'nodejs8'
+  | 'nodejs10'
+  | 'nodejs12'
+  | 'nodejs14'
+  | 'nodejs16'
+  | 'python37'
+  | 'python38'
+  | 'python39'
+  | 'go111'
+  | 'go113'
+  | 'go116'
+  | 'java11'
+  | 'dotnet3'
+  | 'ruby26'
+  | 'ruby27';
+export type CloudFunctionMemory = 128 | 256 | 512 | 1024 | 2048 | 4096;
+export type CloudFunctionVpcEgress = 'ALL' | 'ALL_TRAFFIC' | 'PRIVATE' | 'PRIVATE_RANGES_ONLY';
+export type CloudFunctionRegion =
+  | 'us-central1'
+  | 'us-east1'
+  | 'us-east4'
+  | 'europe-west1'
+  | 'europe-west2'
+  | 'asia-east1'
+  | 'asia-east2'
+  | 'asia-northeast1'
+  | 'asia-northeast2'
+  | 'us-west2'
+  | 'us-west3'
+  | 'us-west4'
+  | 'northamerica-northeast1'
+  | 'southamerica-east1'
+  | 'europe-west3'
+  | 'europe-west6'
+  | 'europe-central2'
+  | 'australia-southeast1'
+  | 'asia-south1'
+  | 'asia-southeast1'
+  | 'asia-southeast2'
+  | 'asia-northeast3';
+export type ServiceName = string;
+
+export interface GCP {
+  configValidationMode?: 'error' | 'warn' | 'off';
+  console?:
+    | boolean
+    | {
+        monitoring?: {
+          logs?: {
+            disabled?: boolean;
+          };
+          request?: {
+            disabled?: boolean;
+          };
+          response?: {
+            disabled?: boolean;
+          };
+        };
+        org?: string;
+      };
+  custom?: {
+    [k: string]: unknown;
+  };
+  dashboard?: {
+    disableMonitoring?: boolean;
+  };
+  deprecationNotificationMode?: 'error' | 'warn' | 'warn:summary';
+  disabledDeprecations?: '*' | ErrorCode[];
+  frameworkVersion?: string;
+  functions?: {
+    /**
+     * This interface was referenced by `undefined`'s JSON-Schema definition
+     * via the `patternProperty` "^[a-zA-Z0-9-_]+$".
+     */
+    [k: string]: {
+      name?: string;
+      events?: (
```

Comment:
I know this was auto generated, but this library throws when `events` is not present. 
>Error: Missing "events" property for function "X". Your function needs at least one "event". Please check the docs for more info.

Reply:
Got the source: https://github.com/serverless/serverless-google-cloudfunctions/blob/b1cdaf25c1cb782b906ea4bfbf0adbd4b4efa4fd/shared/validate.js#L55-L63

Reply:
Good catch! Perhaps these keys should also be added as required in the JSON schema https://github.com/serverless/serverless-google-cloudfunctions/blob/master/provider/googleProvider.js#L151

Reply:
I agree. 

Haven't seen much in the way of maintenance on this repo lately. Hopefully this can get some attention and get in there.

Reply:
It seems adding event's causes a runtime error. Pretty strange though, as many of the other properties should also fall under this boat:

```ts
events: {
  type: 'string',
  items: { $ref: '#/definitions/cloudFunctionEvent' },
},
```

The type of `cloudFunctionEvent` nor the `events` key doesn't seem to matter. Even `string` threw this error.

>Property 'functions[].events' already have a definition - this property might have already been defined by the Serverless framework or one other plugin